README.md

```diff
@@ -122,24 +122,101 @@ For more information regarding the configuration of Open Liberty's logging capab
+        *  **TIP** - If your Infinispan deployment and Liberty deployment are in different namespaces/projects, you will need to set the `INFINISPAN_HOST`, `INFINISPAN_PORT`, `INFINISPAN_USER`, and `INFINISPAN_PASS` environment variables in addition to the `INFINISPAN_SERVICE_NAME` environment variable. This is due to the Liberty deployment not having the access to the Infinispan service environment variables it requires.
+
+    ```dockerfile
+    ### Infinispan Session Caching ###
+    FROM open-liberty AS infinispan-client
+    
+    # Install Infinispan client jars
+    USER root
+    RUN apt-get update \
+        && apt-get install -y --no-install-recommends maven \
+        && mkdir -p /opt/ol/wlp/usr/shared/resources/infinispan \
+        && echo '<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">  <modelVersion>4.0.0</modelVersion>   <groupId>io.openliberty</groupId>  <artifactId>openliberty-infinispan-client</artifactId>  <version>1.0</version>  <!-- https://mvnrepository.com/artifact/org.infinispan/infinispan-jcache-remote -->  <dependencies>    <dependency>      <groupId>org.infinispan</groupId>      <artifactId>infinispan-jcache-remote</artifactId>      <version>10.1.3.Final</version>    </dependency>  </dependencies></project>' > /opt/ol/wlp/usr/shared/resources/infinispan/pom.xml \
+        && mvn -f /opt/ol/wlp/usr/shared/resources/infinispan/pom.xml dependency:copy-dependencies -DoutputDirectory=/opt/ol/wlp/usr/shared/resources/infinispan \
+        && apt-get remove -y maven \
+        && rm -f /opt/ol/wlp/usr/shared/resources/infinispan/pom.xml \
+        && rm -f /opt/ol/wlp/usr/shared/resources/infinispan/jboss-transaction-api*.jar \
+        && rm -f /opt/ol/wlp/usr/shared/resources/infinispan/reactive-streams-*.jar \
+        && rm -f /opt/ol/wlp/usr/shared/resources/infinispan/rxjava-*.jar \
+        && rm -rf ~/.m2 \
+        && chown -R 1001:0 /opt/ol/wlp/usr/shared/resources/infinispan \
+        && chmod -R g+rw /opt/ol/wlp/usr/shared/resources/infinispan
+    USER 1001
+    
+    FROM open-liberty AS open-liberty-infinispan
+    USER root
+    
+    # Copy Infinispan client jars to Open Liberty shared resources
+    COPY --from=infinispan-client /opt/ol/wlp/usr/shared/resources/infinispan /opt/ol/wlp/usr/shared/resources/infinispan
+    
+    # Instruct docker-server.sh to use Infinispan for session caching.
```

Comment:
The best practices for our snippets it to be pushed during build time, via `configure.sh`.  The only exception is part of the certificate manager - but all other snippets following the convention of being copied during build time and configured via runtime.

Reply:
Take a look in `configure.sh` to see other examples.